omnigibson/prims/cloth_prim.py

```diff
@@ -136,6 +179,361 @@ def _post_load(self):
         # Store the default position of the points in the local frame
         self._default_positions = vtarray_to_torch(self.get_attribute(attr="points"))
 
+    def _remesh(self):
+        assert self.prim is not None, "Cannot remesh a non-existent prim!"
+        has_uv_mapping = self.prim.GetAttribute("primvars:st").Get() is not None
+
+        # We will remesh in pymeshlab, but it doesn't allow programmatic construction of a mesh with texcoords so
+        # we convert our mesh into a trimesh mesh, then export it to a temp file, then load it into pymeshlab
+        scaled_world_transform = PoseAPI.get_world_pose_with_scale(self.prim.GetPath().pathString)
+        # Convert to trimesh mesh (in world frame)
+        tm = mesh_prim_to_trimesh_mesh(
+            mesh_prim=self.prim, include_normals=True, include_texcoord=True, world_frame=True
+        )
+        # Tmp file written to: {tmp_dir}/{tmp_fname}/{tmp_fname}.obj
+        tmp_name = str(uuid.uuid4())
+        tmp_dir = os.path.join(tempfile.gettempdir(), tmp_name)
+        tmp_fpath = os.path.join(tmp_dir, f"{tmp_name}.obj")
+        Path(tmp_dir).mkdir(parents=True, exist_ok=True)
+        tm.export(tmp_fpath)
+
+        # Start with the default particle distance
+        particle_distance = self.cloth_system.particle_contact_offset * 2 / 1.5
+
+        # Repetitively re-mesh at lower resolution until we have a mesh that has less than MAX_CLOTH_PARTICLES vertices
+        import pymeshlab  # We import this here because it takes a few seconds to load.
```

Comment:
sorry I am not sure I fully understand this. what benefit do we gain by importing here exactly?

Reply:
I didn't add this, I just moved it over from the micro particle system class. But I'm guessing pymeshlab takes a few seconds to import so the author wanted to save some time by only importing it if remeshing is needed (now VERY infrequently). I guess pymeshlab is actually now an optional dependency so we could even remove it from the setup.py later if needed.